src/servant/tt_001/F84.c

```diff
@@ -109,7 +109,104 @@ s32 UpdateEntityVelocityTowardsTarget(s32 unused, s32 targetX, s32 targetY) {
     return D_us_801735C0;
 }
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_80171284);
+Entity* func_us_80171284(
+    Entity* self) { // Assume self is also an Entity pointer
+    Entity* entity;
+    s32 i;
+    s32 index;
+    u32 matches = 0;
+#if !defined(VERSION_PSP)
+    s32 posDelta;
+#endif
+
+    // Hunt through these entities looking for ones that match all criteria.
+    // Call them a success and increment successes.
+    entity = &D_800762D8[0];
+    for (i = 0; i < 128; i++, entity++) {
+        D_us_801735C4[i] = 0;
+
+        // Very similar code to CheckAllEntitiesValid
+        if (!entity->entityId)
+            continue;
+        if (entity->hitboxState == 0)
+            continue;
+        if (entity->flags & FLAG_UNK_00200000)
+            continue;
+        if (entity->posX.i.hi < -16)
+            continue;
+        if (entity->posX.i.hi > 272)
+            continue;
+        if (entity->posY.i.hi > 240)
+            continue;
+        if (entity->posY.i.hi < 0)
+            continue;
+            // Differs from here
+#if defined(VERSION_PSP)
+        if (entity->hitboxState & 8 &&
+            D_us_80170510[D_us_80173810.level / 10][2] == 0)
```

Comment:
This is likely some sort of struct, but the access is a bit different between PSP and PSX, so I've just made it like this for now - feedback welcome here if there's a better design that can work for both

Reply:
I think D_us_80173810 is a fake symbol and the real one is D_us_80173810-0x8. That should make it match in both places

Reply:
Sorry, tried a few things and couldn't get a match so maybe I'm not understanding exactly what you mean, can you please clarify?

Looks like `D_us_80173810-0x8` (`D_us_8017380C`) is a struct with two s16?
```
glabel D_us_80173808
    /* 3808 80173808 */ .short 0x0000
    /* 380A 8017380A */ .short 0x0000
.size D_us_80173808, . - D_us_80173808
```

Reply:
The data gets split up according to what splat detects and isn't necessarily correct. This doesn't work? https://decomp.me/scratch/sc3CM

Reply:
Yep that worked. The address was already defined and used further on in this file so I think it was conflicting with my other definition in `servant.h` which I've now removed. Thanks!